solana-programs/anchor/audius-data/tests/audius-data.ts

```diff
@@ -882,18 +710,95 @@ describe("audius-data", function () {
       updateUser({
         program,
         metadata: randomCID(),
-        userStorageAccount: newUserAcctPDA,
-        userAuthorityKeypair: userAuthorityDelegateKeypair,
-        userAuthorityDelegate: userAuthorityDelegatePDA,
-        authorityDelegationStatusAccount: authorityDelegationStatusPda,
+        userStorageAccount: delegate.userAccountPDA,
+        userAuthorityKeypair: delegate.userAuthorityDelegateKeypair,
+        userAuthorityDelegate: delegate.userAuthorityDelegatePDA,
+        authorityDelegationStatusAccount: delegate.authorityDelegationStatusPDA,
       })
     )
       .to.eventually.be.rejected.and.property("msg")
       .to.include(`This authority's delegation status is revoked.`);
   });
 
+  it("delegate adds/removes another delegate", async function () {
+    const firstDelegate = await testCreateUserDelegate({
+      adminKeypair,
+      adminStorageKeypair,
+      program,
+      provider,
+      ...getURSMParams(),
+    });
+
+    const secondDelegate = await testCreateUserDelegate({
+      adminKeypair,
+      adminStorageKeypair,
+      program,
+      provider,
+      ...getURSMParams(),
+    });
+
+    const userAuthorityDelegateSeeds = [
+      firstDelegate.userAccountPDA.toBytes().slice(0, 32),
+      secondDelegate.userAuthorityDelegateKeypair.publicKey
+        .toBytes()
+        .slice(0, 32),
+    ];
+    const res = await PublicKey.findProgramAddress(
+      userAuthorityDelegateSeeds,
+      program.programId
+    );
+    const newUserAuthorityDelegatePDA = res[0];
+    const newUserAuthorityDelegateBump = res[1];
+
+    const addUserAuthorityDelegateArgs = {
+      accounts: {
+        admin: adminStorageKeypair.publicKey,
+        user: firstDelegate.userAccountPDA,
+        newUserAuthorityDelegate: newUserAuthorityDelegatePDA,
+        signerUserAuthorityDelegate: firstDelegate.userAuthorityDelegatePDA,
+        authorityDelegationStatus: firstDelegate.authorityDelegationStatusPDA,
+        authority: firstDelegate.userAuthorityDelegateKeypair.publicKey,
+        payer: provider.wallet.publicKey,
+        systemProgram: SystemProgram.programId,
+      },
+      signers: [firstDelegate.userAuthorityDelegateKeypair], // the first user's delegate signs
+    };
+
+    await program.rpc.addUserAuthorityDelegate(
+      firstDelegate.baseAuthorityAccount,
```

Comment:
this test would be clearer if global props like `handleBytesArray`, `userBumpSeed`, `userAccount` didn't come from `firstDelegate` or `secondDelegate`, and props that are particular to a delegate like `userAuthorityDelegateKeypair` were accessed through `firstDelegate.<prop>` etc.